### PR TITLE
Refactor: take blockchain out of MainnetTransactionProcessor

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Remove an unused parameter in MainnetTransactionProcessor 
processTransaction method, as well as all code supporting 
sending that parameter.

This is a leftover parameter from the EVM library refactoring. 
`blockchain` was once used to get info for the `BLOCKHASH` 
operation, replaced by the `BlockHashLookup` interface. 

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Review by macfarla — APPROVED** [review]
LGTM

File: ethereum/referencetests/src/reference-test/java/org/hyperledger/besu/ethereum/vm/GeneralStateReferenceTestTools.java [hunk 1]
```diff
@@ -153,8 +153,7 @@ public static void executeTest(final GeneralStateTestCaseEipSpec spec) {
             .blobGasPricePerGas(blockHeader.getExcessBlobGas().orElse(BlobGas.ZERO));
     final TransactionProcessingResult result =
         processor.processTransaction(
-            blockchain,
-            worldStateUpdater,
+                worldStateUpdater,
```
Comment (macfarla) [inline ethereum/referencetests/src/reference-test/java/org/hyperledger/besu/ethereum/vm/GeneralStateReferenceTestTools.java:156]: this formatting seems out of line but I'm assuming spotless has it under control

**Review by macfarla — COMMENTED** [review]
LGTM

**Review by ahamlat — APPROVED** [review]
LGTM

**Pull request merged**